client/internal/retry/do.go

```diff
@@ -21,6 +22,8 @@ type Options struct {
 	Retries int           // Retries is the max number of retries before terminating. Negative means no limit.
 
 	Logger *log.Logger
+
+	IgnoreError bool // IgnoreError will ignore and save error returned from Func, when max retries exceeded, errors are combined the returned together.
```

Comment:
Added this to improve robustness, because our aegis api is flaky and not eventual consistent

Reply:
I think we should rename this - we don't actually ignore the errors but instead prevent an early termination from them.

What about `PreventEarlyExitOnError`?

Reply:
Yea, make sense, did not think too much when writing it down, let's use `EarlyExitOnError`? `Prevent` seems slightly redundant